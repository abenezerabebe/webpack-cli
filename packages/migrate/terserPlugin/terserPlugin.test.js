"use strict";

const defineTest = require("@webpack-cli/utils/defineTest").default;

defineTest(__dirname, "terserPlugin", "terserPlugin-0");
defineTest(__dirname, "terserPlugin", "terserPlugin-1");
defineTest(__dirname, "terserPlugin", "terserPlugin-2");
defineTest(__dirname, "terserPlugin", "terserPlugin-3");
defineTest(__dirname, "terserPlugin", "terserPlugin-4");
