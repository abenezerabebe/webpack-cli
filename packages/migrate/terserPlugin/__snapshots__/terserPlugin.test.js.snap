// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`terserPlugin transforms correctly using "terserPlugin-0" data 1`] = `
"module.exports = {
	optimization: {
        minimize: true
    }
};
"
`;

exports[`terserPlugin transforms correctly using "terserPlugin-1" data 1`] = `
"const TerserPlugin = require(\\"terser-webpack-plugin\\");
module.exports = {
	devtool: \\"source-map\\",
	optimization: {
        minimize: true,

        minimizer: [new TerserPlugin({
			sourceMap: true,
			compress: {}
		})]
    }
};
"
`;

exports[`terserPlugin transforms correctly using "terserPlugin-2" data 1`] = `
"const TerserPlugin = require(\\"terser-webpack-plugin\\");
module.exports = {
	devtool: \\"source-map\\",
	optimization: {
        minimize: true,

        minimizer: [new TerserPlugin({
			sourceMap: true,
			compress: {}
		})]
    }
};
"
`;

exports[`terserPlugin transforms correctly using "terserPlugin-3" data 1`] = `
"module.exports = {
    devtool: \\"eval\\",
    entry: [\\"./src/index\\"],

    output: {
		path: path.join(__dirname, \\"dist\\"),
		filename: \\"index.js\\"
	},

    module: {
		loaders: [
			{
				test: /.js$/,
				loaders: [\\"babel\\"],
				include: path.join(__dirname, \\"src\\")
			}
		]
	},

    resolve: {
		root: path.resolve(\\"/src\\"),
		modules: [\\"node_modules\\"]
	},

    plugins: [new webpack.optimize.OccurrenceOrderPlugin()],
    debug: true,

    optimization: {
        minimize: true
    }
};
"
`;

exports[`terserPlugin transforms correctly using "terserPlugin-4" data 1`] = `
"const TerserPlugin = require('terser-webpack-plugin');
module.exports = {
    devtool: \\"eval\\",
    entry: [\\"./src/index\\"],

    output: {
		path: path.join(__dirname, \\"dist\\"),
		filename: \\"index.js\\"
	},

    module: {
		loaders: [
			{
				test: /.js$/,
				loaders: [\\"babel\\"],
				include: path.join(__dirname, \\"src\\")
			}
		]
	},

    resolve: {
		root: path.resolve(\\"/src\\"),
		modules: [\\"node_modules\\"]
	},

    plugins: [new webpack.optimize.OccurrenceOrderPlugin()],
    debug: true,

    optimization: {
        minimize: true,

        minimizer: [new TerserPlugin({
			sourceMap: true
		})]
    }
};
"
`;
